roofit/roofitcore/inc/RooAbsData.h

```diff
@@ -360,10 +375,12 @@ class RooAbsData : public TNamed, public RooPrintable {
 
   std::unique_ptr<RooArgSet> _globalObservables; // Snapshot of global observables
 
+  mutable TNamed* _namePtr ; //! De-duplicated name pointer. This will be equal for all objects with the same name.
+
 private:
   void copyGlobalObservables(const RooAbsData& other);
 
-   ClassDef(RooAbsData, 6) // Abstract data collection
+   ClassDef(RooAbsData, 7) // Abstract data collection
```

Comment:
No need to increase class version if you take it out of I/O using `//!`.
```suggestion
   ClassDef(RooAbsData, 6) // Abstract data collection
```